packages/components/src/List/Toolbar/ColumnChooserButton/README.md

```diff
@@ -0,0 +1,274 @@
+
+| Props | Type | Info
+|---|---|---|
+| className | string | class passed to the tooltip header
+| children | React.element, [React.element] | the header's content of the popover
+
+### ColumnChooserFooter
+
+Footer of the column chooser.
+The apply button is here and it submit the form.
+You can pass children and override the default render.
+By default, it consumes the [columnChooserContext](#columnchoosercontext).
+
+The component holds reference :
+* Submit : [SubmitButton](#SubmitButton)
+* SelectAll: [SelectAllCheckbox](#SelectAllCheckbox)
+
+| Props | Type | Info
+|---|---|---|
+| className | string | class passed to the tooltip header
+| children | React.element, [React.element] | the footer  content of the popover
+
+### ColumnChooserBody
+
+Body of the column chooser.
+It renders the different column row.
+You can pass a function as children, it will receive the columns as arguments.
+Else, if children is not a function it will render normally.
+It consumes the [columnChooserContext](#columnchoosercontext).
+
+The component holds reference :
+* Row : [ColumnChooserRowRenderer](#columnchooserrowrenderer)
+
+| Props | Type | Info
+|---|---|---|
+| children | function | the body's  content of the popover, called with the columns as arguments.
+
+### ColumnChooserRowRenderer
+
+The row renderer only render its children.
+It holds reference to other components: , .
+By default, it consumes the [columnChooserContext](#columnchoosercontext).
+
+The component holds reference :
+* Visibility: [RowCheckbox](#rowcheckbox)
+* Label: [RowLabel](#rowlabel)
+
+| Props | Type | Info
+|---|---|---|
+| className | string | class passed to div wrapper row
+| children | React.element, [React.element] | the row's  content of the popover
+
+### RowCheckbox
+
+Add a checkbox to switch the visibility of a column.
+If the column is locked, a locked icon replaces the checkbox.
+	dataFeature,
+	describedby,
+	description,
+	id,
+	label,
+	locked = false,
+	onClick,
+	checked = false,
+
+| Props | Type | Info
+|---|---|---|
+| dataFeature | string | data-feature attribute of the input
+| describedby | string | the id of the markup holding the description
+| description | string | the description
+| id | string | id of the checkbox input
+| label | string | label used
+| locked | bool | indicates if the column is locked or not
+| onChange | func | trigger when the checkbox is clicked
+| checked | bool | indicates the visibility value, and if the checkbox is checked
+
+### RowLabel
+
+Add the label of the column to the row.
+
+| Props | Type | Info
+|---|---|---|
+| label | string | the label of the column
+
+## Hooks
+### ColumnChooserManagerHook
+
+This hook manages all the logic behavior in the [ColumnChooser](#columnchooser).
+It holds the state of the columns edited by the user.
+
+| Parameters | Type | Info
+|---|---|---|
+| initColumns | array | array of columns object from the list
```

Comment:
This info is not that clear. Is it
- the initial columns ?
- the all the columns, updated if it changes ?
- the filtered columns ?

I think the naming is not right. The description  seems to indicate that it's the second choice but the name is disturbing

Reply:
What do you suggest ? These columns are the columns from the list and they will be transform to the columns chooser ones. They are not filtered. And they are used as the initial render to init the hook which hold the logic behind the column chooser.
I'm waiting for your suggestion before any change on naming.